app/components/interview_bookings_component.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+class InterviewBookingsComponent < ViewComponent::Base
+  def initialize(application_choice)
+    @application_choice = application_choice
+  end
+
+  def interviews
+    @application_choice.interviews.includes(:provider).order(date_and_time: :asc)
+  end
+
+  def formatted_time(interview)
+    interview.date_and_time.to_s(:govuk_date_and_time)
+  end
+
+  def provider_name(interview)
+    interview.provider.name
+  end
+
+  def location(interview)
+    make_urls_clickable(
+      show_newlines(
+        interview.location,
+      ),
+    )
+  end
+
+  def additional_details(interview)
+    make_urls_clickable(
+      show_newlines(
+        interview.additional_details,
+      ),
+    )
+  end
+
+private
+
+  def show_newlines(text)
+    simple_format(text, class: 'govuk-body')
+  end
+
+  def make_urls_clickable(text)
+    text
+      .yield_self { |t| t.gsub(URI::DEFAULT_PARSER.make_regexp, '<a href="\0">\0</a>') }
```

Comment:
Do we need to set the allowable schemes here (ie only `http` and `https`) to avoid cross-site-scripting security issues (eg `javascript:alert('test')`)?

Reply:
Good question! The sanitize call underneath handles that case, but I'm going to add these anyway because:

- It's more explicit
- It allows for the refactor in commit https://github.com/DFE-Digital/apply-for-teacher-training/pull/3919/commits/eef9799d85c40ac377a79f2b7d15f9fddea8ff9a